src/main/frontend/src/app/ltistorage.js

```diff
@@ -1,488 +1,248 @@
-        );
-      }
-      return Promise.reject({
-        code: "not_found",
-        message: "Capabilities not found",
-      });
-    });
-  }
-  async putData(key, value) {
-    function secureRandom(length) {
-      let random = new Uint8Array(length||63);
-      crypto.getRandomValues(random);
-      return btoa(String.fromCharCode(...random)).replace(/\//g, '_').replace(/\+/g, '-');
+    constructor(targetOrigin, launchFrame) {
+        this._targetOrigin = new URL(targetOrigin);
+        this._launchFrame = launchFrame || window;
     }
-    return {
-      subject: "lti.put_data",
-      key: key,
-      value: value,
-      message_id: 'message-' + secureRandom(15)
-    };
-  }
-
-  async getData(key) {
-  function secureRandom(length) {
-        let random = new Uint8Array(length||63);
+    static secureRandom(length) {
+        let random = new Uint8Array(length || 63);
         crypto.getRandomValues(random);
         return btoa(String.fromCharCode(...random)).replace(/\//g, '_').replace(/\+/g, '-');
-      };
-    return this.sendPostMessageIfCapable({
-      subject: "lti.get_data",
-      key: key,
-      message_id: 'message-' + secureRandom(15)
-    }).then((response) => {
-      return response.value;
-    });
-  }
-}
-
-class LtiPostMessageLog {
-  constructor(debug) {
-    _LtiPostMessageLog_debug.set(this, false);
-    this._request = {};
-    this._response = {};
-    this._error = [];
-    this._start_time = Date.now();
-    __classPrivateFieldSet(this, _LtiPostMessageLog_debug, debug, "f");
-  }
+    }
+    async sendPostMessage(data, targetWindow, originOverride, targetFrameName) {
+        return new Promise((resolve, reject) => {
+            let timeout;
+            let targetOrigin = originOverride || this._targetOrigin.origin;
+            let targetFrame;
+            try {
+                targetFrame = this.getTargetFrame(targetWindow, targetFrameName);
+            }
+            catch (e) {
+                targetFrameName = undefined;
+                targetFrame = targetWindow;
+            }
+            const messageHandler = (event) => {
+                if (event.data.message_id !== data.message_id) {
+                    //log.error({message: 'Ignoring message, invalid message_id: [' + event.data.message_id + '] expected: [' + data.message_id + ']'});
+                    return;
+                }
+                /*
+                log.response(event);
+                if (targetOrigin !== '*' && event.origin !== targetOrigin) {
+                    log.error({message: 'Ignoring message, invalid origin: ' + event.origin});
+                    return log.print();
+                }
+                if (event.data.subject !== data.subject + '.response') {
+                    log.error({message: 'Ignoring message, invalid subject: [' + event.data.subject + '] expected: [' + data.subject + '.response]'});
+                    return log.print();
+                }
+                */
```

Comment:
Do we want to remove these, I feel like the checks are quite useful but since we stripped the logging out of this it wouldn't do anything. I wondered if there was a way we could use these checks but do something else? There are a few other `log.error` around this code too that have been commented out for now since they wouldn't work, but again I felt leaving them in is a handy reference for now. 

Let me know your thoughts!

Reply:
Since we have access to the original library, I'd remove this and other commented code.